clients/drcachesim/tracer/tracer.cpp

```diff
@@ -918,8 +949,30 @@ instrument_memref(void *drcontext, user_data_t *ud, instrlist_t *ilist, instr_t
     }
     if (op_L0_filter.get_value())
         insert_load_buf_ptr(drcontext, ilist, where, reg_ptr);
-    adjust = instru->instrument_memref(drcontext, ilist, where, reg_ptr, adjust, app, ref,
-                                       ref_index, write, pred);
+
+#ifdef AARCH64
+    // TODO i#4400: Use a new operand type to back-align zeroed address, instead of
+    // doing it in drcachesim.
+    if (is_dc_zva_instr(app)) {
+        reg_id_t reg_ba_addr;
+        if (drreg_reserve_register(drcontext, ilist, where, NULL, &reg_ba_addr) !=
```

Comment:
This is failing to use the predicate set inside instru->instrument_memref.  I suppose A64 has no predicates, but it feels like it's not the model way to do it: someone will mirror this for some A32 code and fail to notice the bug, assuming this code here has no flaws.  Plus, if it could be elided, this skips all of that code: with the same arguments.  It seems best to be inside the instru_t classes and not out here.